my_init.d/create-system-user.sh

```diff
@@ -1,17 +1,27 @@
 #!/bin/bash
 set -em
 
-# Add $SYSTEM_USER user (no password)
+# Add $SYSTEM_USER user (no password).
 mkdir /home/$SYSTEM_USER                                                         && \
       useradd --home /home/$SYSTEM_USER --uid $SYSTEM_USER_UID --shell /bin/bash $SYSTEM_USER && \
       chown -R $SYSTEM_USER:$SYSTEM_USER /home/$SYSTEM_USER
 
-# Add .bashrc file to the $SYSTEM_USER user
+# Add .bashrc file to the $SYSTEM_USER user.
 cp -v /etc/skel/.bashrc /etc/skel/.bash_logout /etc/skel/.profile /home/$SYSTEM_USER/
 
-echo "export PYTHONPATH=\"/home/$SYSTEM_USER\"" >> /home/$SYSTEM_USER/.bashrc
+# Update .bashrc file.
 echo "export PATH=$PATH:\"/home/$SYSTEM_USER/.local/bin\"" >> /home/$SYSTEM_USER/.bashrc
+grep "reentry scan" /home/$SYSTEM_USER/.bashrc || echo "reentry scan" >> /home/$SYSTEM_USER/.bashrc
 
-mkdir --mode=0700 /home/$SYSTEM_USER/.ssh/ &&  touch /home/$SYSTEM_USER/.ssh/known_hosts
+
+# Create .ssh folder with an check that known_hosts file is present there
+mkdir --mode=0700 /home/$SYSTEM_USER/.ssh/
```

Comment:
won't this fail when the folder is already there?

perhaps it would be best to actually test this on travis, i.e.:
start the container, stop the container, start it again, and then run whatever tests we want

Reply:
hm, I guess one would need to mount a persistent home directory for this test (could be a local /tmp folder)

Reply:
well, if you start-stop the container things will be erased anyway.

Reply:
it wouldn't be difficult to add and we should do it (I guarantee it will save us time), but it doesn't have to be in this PR

Reply:
I think it is better do it in a separate PR. There is a related issue that I am currently thinking about. So I am going to add the issue and make the PR that fixes it.

Reply:
This is the issue: https://github.com/aiidateam/aiida-prerequisites/issues/10

Reply:
as mentioned below, I added such a check